plugins/BTagAnalyzer.cc

```diff
@@ -1009,353 +1024,358 @@ void BTagAnalyzerT<IPTI,VTX>::analyze(const edm::Event& iEvent, const edm::Event
-	  EventInfo.nDHadrons++;
-	} // final c hadron
-      } // c hadron
-
-      // Leptons
-      if ( (ID == 11 || ID == 13 || ID == 15) && genIt.p4().pt() > 3. ) {
-       if (genIt.numberOfMothers()>0) {  // protection for herwig ttbar mc
-        const Candidate * moth1 = genIt.mother();
-        if ( moth1->pdgId() != genIt.pdgId() ) {
-          EventInfo.Genlep_pT[EventInfo.nGenlep]    = genIt.p4().pt();
-          EventInfo.Genlep_eta[EventInfo.nGenlep]   = genIt.p4().eta();
-          EventInfo.Genlep_phi[EventInfo.nGenlep]   = genIt.p4().phi();
-          EventInfo.Genlep_pdgID[EventInfo.nGenlep] = genIt.pdgId();
-          EventInfo.Genlep_status[EventInfo.nGenlep] = genIt.status();
-          int ID1 = abs( moth1->pdgId() );
-          int ID2 = -9999;
-          int ID3 = -9999;
-          int ID4 = -9999;
-          int isWZ = 0, istau = 0, isB = 0, isD = 0;
-          if (moth1->numberOfMothers()>0) {   // protection for herwig ttbar mc
-            const Candidate * moth2 = moth1->mother();
-            ID2 = abs( moth2->pdgId() );
-            if (moth2->numberOfMothers()>0) {
-              const Candidate * moth3 = moth2->mother();
-              ID3 = abs( moth3->pdgId() );
-              if (moth3->numberOfMothers()>0) {
-                const Candidate * moth4 = moth3->mother();
-                ID4 = abs( moth4->pdgId() );
+
+      if(storeGenVariables_){
+        // Leptons
+        if ( (ID == 11 || ID == 13 || ID == 15) && genIt.p4().pt() > 3. ) {
+          if (genIt.numberOfMothers()>0) {  // protection for herwig ttbar mc
+            const Candidate * moth1 = genIt.mother();
+            if ( moth1->pdgId() != genIt.pdgId() ) {
+              EventInfo.Genlep_pT[EventInfo.nGenlep]    = genIt.p4().pt();
+              EventInfo.Genlep_eta[EventInfo.nGenlep]   = genIt.p4().eta();
+              EventInfo.Genlep_phi[EventInfo.nGenlep]   = genIt.p4().phi();
+              EventInfo.Genlep_pdgID[EventInfo.nGenlep] = genIt.pdgId();
+              EventInfo.Genlep_status[EventInfo.nGenlep] = genIt.status();
+              int ID1 = abs( moth1->pdgId() );
+              int ID2 = -9999;
+              int ID3 = -9999;
+              int ID4 = -9999;
+              int isWZ = 0, istau = 0, isB = 0, isD = 0;
+              if (moth1->numberOfMothers()>0) {   // protection for herwig ttbar mc
+                const Candidate * moth2 = moth1->mother();
+                ID2 = abs( moth2->pdgId() );
+                if (moth2->numberOfMothers()>0) {
+                  const Candidate * moth3 = moth2->mother();
+                  ID3 = abs( moth3->pdgId() );
+                  if (moth3->numberOfMothers()>0) {
+                    const Candidate * moth4 = moth3->mother();
+                    ID4 = abs( moth4->pdgId() );
+                  }
+                }
               }
+              if ( ID1 == 15 ) istau = 1;
```

Comment:
These seem to be indentation-only changes. They are ~OK with me but they make the review much harder. Please, in the future refrain from them or put all of them in a separate PR, at least we know that no content was actually changed

Reply:
I'm very sorry for this! I put all of them into a separate commit, but I see now, that this doesn't really help when merging a complete PR. Won't happen again.